File: src/main/java/one/oktw/i18n/api/service/TranslationService.java
package one.oktw.i18n.api.service;

import org.jetbrains.annotations.NotNull;
import org.spongepowered.api.entity.living.player.Player;
import org.spongepowered.api.item.inventory.ItemStack;
import org.spongepowered.api.text.Text;
import org.spongepowered.api.text.translation.Translation;

import java.util.Locale;

public interface TranslationService {

    @NotNull
    Text fromItem(@NotNull ItemStack var1);

    @NotNull
    Text ofLiteralPlaceHolder(@NotNull Object... var1);

    @NotNull
    Text ofPlaceHolder(@NotNull Object... var1);

    @NotNull
    Text ofPlaceHolder(@NotNull String var1, @NotNull Object... var2);

    @NotNull
    Text ofPlaceHolderUnscoped(@NotNull String var1, @NotNull Object... var2);

    @NotNull
    Text of(@NotNull String var1, @NotNull Object... var2);

    @NotNull
    Text ofUnscoped(@NotNull String var1, @NotNull Object... var2);

    @NotNull
    Translation translation(@NotNull String var1);

    @NotNull
    Translation translationUnscoped(@NotNull String var1);

    @NotNull
    String toPlain(@NotNull Player var1, @NotNull Text var2);

    @NotNull
    String toPlain(@NotNull Locale var1, @NotNull Text var2);

    @NotNull
    String toLegacy(@NotNull Player var1, @NotNull Text var2);

    @NotNull
    String toLegacy(@NotNull Locale var1, @NotNull Text var2);

    @NotNull
    String toSponge(@NotNull Player var1, @NotNull Text var2);

    @NotNull
    String toSponge(@NotNull Locale var1, @NotNull Text var2);

    @NotNull
    Text removeStyle(@NotNull Text var1);
}

File: src/main/kotlin/one/oktw/i18n/api/service/TranslationService.kt
package one.oktw.i18n.api.service

import org.spongepowered.api.entity.living.player.Player
import org.spongepowered.api.item.inventory.ItemStack
import org.spongepowered.api.text.Text
import org.spongepowered.api.text.translation.Translation
import java.util.*

interface TranslationService {
    // get real item name of an ItemStack
    fun fromItem(item: ItemStack): Text

    // place holders
    fun ofLiteralPlaceHolder(vararg values: Any): Text
    fun ofPlaceHolder(vararg values: Any): Text
    fun ofPlaceHolder(key: String, vararg values: Any): Text
    fun ofPlaceHolderUnscoped(key: String, vararg values: Any): Text

    // normal Texts
    fun of(key: String, vararg values: Any): Text
    fun ofUnscoped(key: String, vararg values: Any): Text

    // translations
    fun translation(key: String): Translation
    fun translationUnscoped(key: String): Translation

    // serializers
    fun toPlain(player: Player, text: Text): String
    fun toPlain(locale: Locale, text: Text): String
    fun toLegacy(player: Player, text: Text): String
    fun toLegacy(locale: Locale, text: Text): String
    fun toSponge(player: Player, text: Text): String
    fun toSponge(locale: Locale, text: Text): String
}